misc/mdns/src/service.rs

```diff
@@ -147,21 +158,27 @@ impl MdnsService {
         Ok(MdnsService {
             socket,
             query_socket: UdpSocket::bind((Ipv4Addr::from([0u8, 0, 0, 0]), 0u16)).await?,
-            query_interval: Interval::new(Duration::from_secs(20)),
+            query_interval: Interval::new_at(Instant::now(), Duration::from_secs(20)),
```

Comment:
Why change this?

Reply:
Simply to not have to wait for 20 seconds within the unit tests. Other than adding a bit of complexity I don't see it having an impact at the compiled logic.

The same is done in current *master*: https://github.com/libp2p/rust-libp2p/blob/46c62b0145a45c899c4557dc5ac9512f779ae8f4/misc/mdns/src/service.rs#L161